### PR TITLE
Should have a duration for the workoutView now

### PR DESCRIPTION
**Pull request by WettyWednesday**

### CODE REVIEW

File: src/main/java/Group15/Model/Workout.java [hunk 1]
```diff
@@ -10,12 +10,26 @@
 
 public class Workout implements Serializable {
     private final List<WorkoutExercise> exercises = new ArrayList<>();
+    public static final int BREAK_BETWEEN_SETS = 180000;
```
Comment (emilsavino) [inline src/main/java/Group15/Model/Workout.java:13]: isn't this quite a long break?

File: src/main/java/Group15/Util/WorkoutAlgorithm.java [hunk 2]
```diff
@@ -36,36 +33,39 @@ public static Workout createWorkoutFromExercises(
...
             int timePerSet = validExercise.timePerRep * 5;
 
-            while (timeLeftInMilli > 0 && workoutExercise.getSets() < 3)
-            {
-                timeLeftInMilli -= timePerSet;
+            while (initialTimeInMilli - workout.getWorkoutDuration() > 0 && workoutExercise.getSets() < MAX_SETS) {
                 workoutExercise.setSets(workoutExercise.getSets() + 1);
-                timeLeftInMilli -= breakBetweenSets;
+
+                int timeUsedThisSet = workout.getWorkoutDuration() + timePerSet;
```
Comment (emilsavino) [inline src/main/java/Group15/Util/WorkoutAlgorithm.java:39]: This is does not appear to be the time used this set but time used in total up to that point.
Plus you are calling getWorkoutDuration a few times here, which is not very efficient as the method has to iterate through the whole workout every time.

**Review by emilsavino — CHANGES_REQUESTED** [review]
Also something seems to be wrong as i just got a crazy amount of of exercises for a 30 minute workout

File: src/main/java/Group15/Model/Workout.java (hunk 1 above)
Comment (ThePhoneBeLinging) [reply src/main/java/Group15/Model/Workout.java:13]: Its in milliseconds :)

Comment (emilsavino) [reply src/main/java/Group15/Model/Workout.java:13]: i know but its 3 minutes

Comment (ThePhoneBeLinging) [reply src/main/java/Group15/Model/Workout.java:13]: Fred said 3 minutes

**Review by ThePhoneBeLinging — APPROVED** [review]

**Review by emilsavino — APPROVED** [review]

**Pull request merged**